scipy/stats/_morestats.py

```diff
@@ -1135,7 +1135,14 @@ def _boxcox_inv_lmbda(x, y):
     return np.real(-num / np.log(x) - 1 / y)
 
 
-def boxcox_normmax(x, brack=None, method='pearsonr', optimizer=None):
+class BigFloat:
+    def __repr__(self):
+        return "BIG_FLOAT"
+
+
+def boxcox_normmax(
+    x, brack=None, method='pearsonr', optimizer=None, *, ymax=BigFloat()
```

Comment:
Response to 3: Previously I indeed use `ymax=None` as the default, but it's not clear and descriptive to the user. Please ref https://github.com/scipy/scipy/pull/19631#discussion_r1414500609

Reply:
All right, well I respectfully disagree about that for the reasons above, but it's not my call.

Reply:
1. Yes, the user can add attributes to the `BigFloat` instance. I'm not seeing how it would change the behavior of the code.
2. Oops `class BigFloat` should be `class _BigFloat`. That said, forgetting an underscore does not make something public. This is explicitly stated in the docs.
3. Yeah, I suggested something more descriptive. Another problem with `None` is that makes it hard to deprecate the argument if we ever need to:  it can be passed by the user and we can't tell the difference between that and a default. So it is not uncommon to use a bare-bones object. Please suggest a new `__repr__`, though.
4. Yes, https://scipy.github.io/devdocs/dev/missing-bits.html#coding-style-and-guidelines

Reply:
In scikit-learn, we often allow, next to numerical values, some special string values with clear meaning, e.g. `ymax="big_float"` and then, inside the function set it accordingly.